pulp_container/app/tasks/builder.py

```diff
@@ -0,0 +1,135 @@
+from pulp_container.app.models import (
+    Blob,
+    BlobManifest,
+    ContainerRepository,
+    Manifest,
+    Tag
+)
+from pulp_container.constants import MEDIA_TYPE
+from pulpcore.app.models.content import Artifact, ContentArtifact
+from pulpcore.plugin.tasking import WorkingDirectory
+
+import json
+import os
+import shutil
+import subprocess
+from uuid import uuid4
+
+
+def get_or_create_blob(layer_json, manifest, path):
+    """
+    Creates Blob from json snippet of manifest.json
+
+    Args:
+        layer_json (json): json
+        manifest (class:`pulp_container.app.models.Manifest`): The manifest
+        path (str): Path of the directory that contains layer
+
+    Returns:
+        class:`pulp_container.app.models.Blob`
+
+    """
+    try:
+        blob = Blob.objects.get(digest=layer_json['digest'])
+    except Blob.DoesNotExist:
+        layer_file_name = "{}{}".format(path, layer_json['digest'][7:])
+        layer_artifact = Artifact.init_and_validate(layer_file_name)
+        layer_artifact.save()
+        blob = Blob(digest=layer_json['digest'], media_type=layer_json['mediaType'])
+        blob.save()
+        ContentArtifact(artifact=layer_artifact, content=blob,
+                        relative_path=layer_json['digest']).save()
+    BlobManifest(manifest=manifest, manifest_blob=blob).save()
```

Comment:
this line will still make a relation in case it is a config blob. what about passing a boolean that will indicate whether it is a regular blob or not? `get_or_create_blob(manifest_json['config'], manifest, path, config_blob=True`) or another solution would be to check the mediatype and compare if it is MEDIA_TYPE.CONFIG_BLOB_OCI

Reply:
I am going to check the media type.